deps/rabbit/test/peer_discovery_tmp_hidden_node_SUITE.erl

```diff
@@ -0,0 +1,379 @@
+                     %% especially that it doesn't rely or create a connection
+                     %% between the two nodes.
+                     connection => standard_io,
+                     wait_boot => infinity},
+    TestEbin = filename:dirname(code:which(?MODULE)),
+    Args0 = maps:get(args, PeerOptions1, []),
+    Args1 = ["-pa", TestEbin | Args0],
+    Env0 = maps:get(env, PeerOptions1, []),
+    Env1 = [{"ERL_LIBS", os:getenv("ERL_LIBS")} | Env0],
+    PeerOptions2 = PeerOptions1#{args => Args1,
+                                 env => Env1},
+    start_test_nodes(Testcase, 1, NodeCount, PeerOptions2, #{}).
+
+start_test_nodes(Testcase, NodeNumber, NodeCount, PeerOptions, Peers)
+  when NodeNumber =< NodeCount ->
+    PeerName0 = rabbit_misc:format("~s-~b", [Testcase, NodeNumber]),
+    PeerOptions1 = PeerOptions#{name => PeerName0},
+    ct:pal("Starting peer with options: ~p", [PeerOptions1]),
+    case catch peer:start(PeerOptions1) of
+        {ok, PeerPid, PeerName} ->
+            ct:pal("Configuring peer '~ts'", [PeerName]),
+            setup_test_node(PeerPid, PeerOptions1),
+            Peers1 = Peers#{PeerName => PeerPid},
+            start_test_nodes(
+              Testcase, NodeNumber + 1, NodeCount, PeerOptions, Peers1);
+        Error1 when not is_map_key(host, PeerOptions1) ->
+            ct:pal("Failed to started peer node:~n"
+                   "Options: ~p~n"
+                   "Error: ~p", [PeerOptions1, Error1]),
+            %% At least when running from a Buildbuddy CI worker, the network
+            %% configuration is incomplete and the host lacks an FQDN. This
+            %% breaks the start of an Erlang node with long names.
+            %%
+            %% To work around that, we mess with the network configuration (the
+            %% Erlang node runs as root) and try to determine a hostname we can
+            %% use. We then try again to start the node.
+            case determine_hostname(PeerOptions1) of
+                {ok, Host} ->
+                    PeerOptions2 = PeerOptions1#{host => Host},
+                    start_test_nodes(
+                      Testcase, NodeNumber, NodeCount, PeerOptions2, Peers);
+                {error, _}  = Error2 ->
+                    ct:pal("Failed to determine a usable hostname:~n"
+                           "Options: ~p~n"
+                           "Error: ~p", [PeerOptions1, Error2]),
+                    stop_test_nodes(Peers),
+                    erlang:throw(Error2)
+            end;
+        Error ->
+            ct:pal("Failed to started peer node:~n"
+                   "Options: ~p~n"
+                   "Error: ~p", [PeerOptions1, Error]),
+            stop_test_nodes(Peers),
+            erlang:throw(Error)
+    end;
+start_test_nodes(_Testcase, _NodeNumber, _Count, _PeerOptions, Peers) ->
+    ct:pal("Peers: ~p", [Peers]),
+    Peers.
+
+determine_hostname(PeerOptions) ->
```

Comment:
Rather than modifying the system `/etc/hosts` file, would it possibly be easier to use the `inetrc` file itself?

https://www.erlang.org/doc/apps/erts/inet_cfg#user-configuration-example

Reply:
It would interfere with the testscases actually testing the `inetrc` file option.